sktime/markov/msm/koopman_reweighted_msm.py

```diff
@@ -130,67 +142,74 @@ class OOMReweightedMSM(_MSMBaseEstimator):
+        return self._model
 
     def fit(self, dtrajs, **kw):
         # remove last lag steps from dtrajs:
         dtrajs_lag = [traj[:-self.lagtime] for traj in dtrajs]
-        count_model = TransitionCountEstimator(lagtime=self.lagtime, mincount_connectivity=self.connectivity_threshold,
-                                               count_mode=self.count_mode).fit(dtrajs).fetch_model()
+
+        # statistics are collected over full trajectories
+        histogram = count_states(dtrajs, ignore_negative=True)
+        # because of double counting, only count lagged trajs
+        count_matrix = TransitionCountEstimator.count(count_mode=self.count_mode, dtrajs=dtrajs_lag,
+                                                      lagtime=self.lagtime, sparse=self.sparse)
+        count_model = TransitionCountModel(count_matrix, counting_mode=self.count_mode, lagtime=self.lagtime,
+                                           state_histogram=histogram)
+        count_model = count_model.submodel_largest(connectivity_threshold=self.connectivity_threshold, directed=True)
 
         # Estimate transition matrix using re-sampling:
-        if self.rank_Ct == 'bootstrap_counts':
-            Ceff_full = effective_count_matrix(dtrajs_lag, self.lagtime)
-            Ceff = submatrix(Ceff_full, count_model.active_set)
-            smean, sdev = bootstrapping_count_matrix(Ceff, nbs=self.nbs)
+        if self.rank_mode == 'bootstrap_counts':
+            effective_count_mat = effective_count_matrix(dtrajs_lag, self.lagtime)
+            Ceff = submatrix(effective_count_mat, count_model.state_symbols)
+            smean, sdev = _impl.bootstrapping_count_matrix(Ceff, nbs=self.nbs)
         else:
-            smean, sdev = bootstrapping_dtrajs(dtrajs_lag, self.lagtime, count_model.n_states, nbs=self.nbs,
-                                               active_set=count_model.active_set)
+            smean, sdev = _impl.bootstrapping_dtrajs(dtrajs_lag, self.lagtime, count_model.n_states_full, nbs=self.nbs,
+                                                     active_set=count_model.state_symbols)
         # Estimate two step count matrices:
-        C2t = twostep_count_matrix(dtrajs, self.lagtime, count_model.n_states)
+        c2t = _impl.twostep_count_matrix(dtrajs, self.lagtime, count_model.n_states_full)
         # Rank decision:
-        rank_ind = rank_decision(smean, sdev, tol=self.tol_rank)
+        rank_ind = _impl.rank_decision(smean, sdev, tol=self.tol_rank)
         # Estimate OOM components:
-        Xi, omega, sigma, eigenvalues = oom_components(count_model.count_matrix.toarray(), C2t, rank_ind=rank_ind,
-                                             lcc=count_model.active_set)
+        if issparse(count_model.count_matrix_full):
+            cmat = count_model.count_matrix_full.toarray()
+        else:
+            cmat = count_model.count_matrix_full
+        Xi, omega, sigma, eigenvalues = _impl.oom_components(cmat, c2t, rank_ind=rank_ind,
```

Comment:
I would love to see these greek letter variables (and `P` below) renamed to something more readable, or for an equation to be explicitly included (let me know if i am missing it)

Reply:
me too. for this I have to dig into OOM MSM theory first, though.

Reply:
no problem if you want to do it later, can always make an issue and merge this anyway